### PR TITLE
Update deprecated lifecycle methods

### PR DESCRIPTION
**Pull request by jkozol**
In React 17 componentWillMount, componentWillUpdate, and componentWillReceieveProps are deprecated. The functionality triggered by these hooks has been updated and moved to other methods.

### CODE REVIEW

File: components/ListView/ListItemComponents.js [hunk 1]
```diff
@@ -10,22 +10,9 @@ class ListItemComponents extends React.Component {
...
-    const olditem = this.props.listItem.name;
-    const newitem = newProps.listItem.name;
-    const parent = this.props.listItemParent;
-    if (olditem !== newitem) {
-      if ($(`.${parent} [data-component="${newitem.name}"]`).hasClass("active")) {
-        this.setState({ expanded: true });
-      } else {
-        this.setState({ expanded: false });
-      }
-    }
```
Comment (jkozol) [inline components/ListView/ListItemComponents.js]: @jgiardino I believe when the props update (without a remount) the state won't be cleared. So, if the component has a new name and was already set as expanded it will remain expanded without this. Please let me know if I misconstrued the purpose of this method.

**Comment by henrywang** [comment]
@martinpitt, only Travis CI got triggered on this PR, any issues about our CI? Thanks?

**Comment by martinpitt** [comment]
@henrywang : Yes, the pods crash, I'll look at this now.

File: components/ListView/DependencyListView.js [hunk 2]
```diff
@@ -4,9 +4,7 @@ import PropTypes from "prop-types";
 import ListView from "./ListView";
 import ListItemComponents from "./ListItemComponents";
 
-class DependencyListView extends React.Component {
-  componentWillMount() {}
-
+class DependencyListView extends React.PureComponent {
```
Comment (KKoukiou) [inline components/ListView/DependencyListView.js]: You can also use the arrow functions for that.
I am not suggesting it, just so that it's clear.

File: components/Pagination/Pagination.js [hunk 3]
```diff
@@ -19,26 +19,29 @@ const messages = defineMessages({
...
+  componentDidUpdate() {
     if (this.paginationPage === document.activeElement) {
       this.paginationPage.select();
     }
   }
 
+  static getDerivedStateFromProps(newProps, prevState) {
+    if (prevState.pageValue !== newProps.currentPage) {
+      return { pageValue: newProps.currentPage };
+    } else return null;
```
Comment (KKoukiou) [inline components/Pagination/Pagination.js]: Does this work correctly? 
I would expect something like:
  return {}
for zero updates.

File: components/Toolbar/FilterInput.js [hunk 4]
```diff
@@ -3,8 +3,8 @@ import PropTypes from "prop-types";
 import { Filter } from "patternfly-react";
 
 class FilterInput extends React.Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
```
Comment (KKoukiou) [inline components/Toolbar/FilterInput.js]: How is this change related?

File: components/ListView/DependencyListView.js (hunk 2 above)
Comment (jkozol) [reply components/ListView/DependencyListView.js]: Ahhh so use `const DependencyListView = () => {...}` instead of a PureComponent?

Comment (KKoukiou) [reply components/ListView/DependencyListView.js]: You can, just stick to some of the two patterns for the whole project basically :)

File: components/Pagination/Pagination.js (hunk 3 above)
Comment (jkozol) [reply components/Pagination/Pagination.js]: I believe returning {} would not cause any updates to the state but per the docs:

> It should return an object to update the state, or null to update nothing

File: components/Toolbar/FilterInput.js (hunk 4 above)
Comment (jkozol) [reply components/Toolbar/FilterInput.js]: I had been considering setting state from props in the constructor and needed this then but you're correct I don't any more. But, I was reading about why props need to be passed to the constructor and super and came across [this section of the docs](https://reactjs.org/docs/state-and-lifecycle.html#adding-local-state-to-a-class) which says to always bind props to the constructor and super. It doesn't provide a reason but I think it is so this.props is initialized by the constructor and not after it. What are your thoughts?

Comment (KKoukiou) [reply components/Toolbar/FilterInput.js]: Yes, basically if you don't call super(props) in the contructor, this.props would still be undefined between the super call and the end of your constructor. But then react does this this.props = props anyway, so you can access the props from render and other methods later on.
So coming back to this specific change, I don't think it's anyhow adding some functionality. :)

File: components/Pagination/Pagination.js (hunk 3 above)
Comment (KKoukiou) [reply components/Pagination/Pagination.js]: Ok thanks

File: components/Toolbar/FilterInput.js (hunk 4 above)
Comment (jkozol) [reply components/Toolbar/FilterInput.js]: Okay I have removed it

File: components/ListView/ListItemComponents.js (hunk 1 above)
Comment (jgiardino) [reply components/ListView/ListItemComponents.js]: This isn't needed. I was looking at this yesterday, and also ended up pulling this out as part of #802.
